### PR TITLE
Fix Coverity issues

### PR DESCRIPTION
**Pull request by CuriousPanCake**
Fix Coverity issues

Fix the following issues reported by Coverity:
* Overflowed constant
* Explicit null dereferenced
* Dereference null return value
* Using invalid iterator

### Tickets:
- CVS-145094

Signed-off-by: Andrii Staikov <andrii.staikov@intel.com>

### CODE REVIEW

File: src/common/snippets/src/lowered/pass/validate_unified_loops.cpp [hunk 1]
```diff
@@ -54,10 +54,12 @@ bool ValidateUnifiedLoops::run(LinearIR& linear_ir) {
...
-                OPENVINO_ASSERT(*dim_indexes.rbegin() == dim_idx,
-                                "Incorrect Loop ID configuration: the Loops with splitted dimension should be successively nested");
-                OPENVINO_ASSERT(loop_manager->get_loop_info(loop_ids[i - 1])->get_increment() == loop_manager->get_loop_info(id)->get_work_amount(),
-                                "Incorrect Loop ID configuration: the Loops with splitted dimension should be successively nested");
+                if (i != 0) {
+                    OPENVINO_ASSERT(*dim_indexes.rbegin() == dim_idx,
+                                    "Incorrect Loop ID configuration: the Loops with splitted dimension should be successively nested");
+                    OPENVINO_ASSERT(loop_manager->get_loop_info(loop_ids[i - 1])->get_increment() == loop_manager->get_loop_info(id)->get_work_amount(),
+                                    "Incorrect Loop ID configuration: the Loops with splitted dimension should be successively nested");
+                }
```
Comment (CuriousPanCake) [inline src/common/snippets/src/lowered/pass/validate_unified_loops.cpp:62]: To be discussed with @a-sidorova

File: src/common/transformations/src/transformations/common_optimizations/fuse_rotary_positional_embeddings.cpp [hunk 2]
```diff
@@ -717,8 +717,15 @@ ov::pass::RoPEShareCosSin::RoPEShareCosSin() {
...
         }
+
         auto it = pattern_map.find(const_inv_freq);
+        if (it == pattern_map.end()) {
+            return false;
+        }
         auto cur_inv_freq = std::dynamic_pointer_cast<opset1::Constant>(it->second.get_node_shared_ptr());
+        if (!cur_inv_freq) {
+            return false;
+        }
```
Comment (CuriousPanCake) [inline src/common/transformations/src/transformations/common_optimizations/fuse_rotary_positional_embeddings.cpp:728]: Can't we just do 

```
auto cur_inv_freq = std::dynamic_pointer_cast<opset1::Constant>(pattern_map[const_inv_freq].get_node_shared_ptr());
if (!cur_inv_freq) {
     return false;
}
```

The reason I haven't done that yet is it is here probably for a reason and the pattern is created using some unusual functionality.

File: src/common/transformations/src/transformations/common_optimizations/lstm_cell_fusion.cpp [hunk 3]
```diff
@@ -328,7 +328,7 @@ ov::pass::LSTMCellFusionWithJointWeights::LSTMCellFusionWithJointWeights() {
 
         std::string f_activation_name = ft->get_type_name();
 
-        if (f_activation_name != it->get_type_name() || f_activation_name != ot->get_type_name())
+        if (!it || f_activation_name != it->get_type_name() || !ot || f_activation_name != ot->get_type_name())
             return false;
 
         f_activation_name[0] = std::tolower(f_activation_name[0]);
```
Comment (CuriousPanCake) [inline src/common/transformations/src/transformations/common_optimizations/lstm_cell_fusion.cpp:334]: @evkotov , can you please review this part of code related to LSTM?

File: src/common/low_precision_transformations/src/pad.cpp [hunk 4]
```diff
@@ -225,6 +225,10 @@ bool PadTransformation::canBeTransformed(const TransformationContext& context, s
                 }
             }
 
+            if ((beginNonZeroIdx == -1) && (endNonZeroIdx == -1)) {
+                return true;
+            }
+
```
Comment (CuriousPanCake) [inline src/common/low_precision_transformations/src/pad.cpp:231]: Discussed with @v-Golubev

**Review by v-Golubev — APPROVED** [review]
LPT part LGTM

File: src/common/snippets/src/lowered/pass/validate_unified_loops.cpp [hunk 5]
```diff
@@ -53,11 +53,13 @@ bool ValidateUnifiedLoops::run(LinearIR& linear_ir) {
...
-                                "Incorrect Loop ID configuration: the Loops with splitted dimension should be successively nested");
-                OPENVINO_ASSERT(loop_manager->get_loop_info(loop_ids[i - 1])->get_increment() == loop_manager->get_loop_info(id)->get_work_amount(),
-                                "Incorrect Loop ID configuration: the Loops with splitted dimension should be successively nested");
+            if (i > 0) {
+                if (std::find(dim_indexes.cbegin(), dim_indexes.cend(), dim_idx) != dim_indexes.cend()) {
+                        OPENVINO_ASSERT(*dim_indexes.rbegin() == dim_idx,
+                                        "Incorrect Loop ID configuration: the Loops with splitted dimension should be successively nested");
+                        OPENVINO_ASSERT(loop_manager->get_loop_info(loop_ids[i - 1])->get_increment() == loop_manager->get_loop_info(id)->get_work_amount(),
+                                        "Incorrect Loop ID configuration: the Loops with splitted dimension should be successively nested");
+                }
```
Comment (CuriousPanCake) [inline src/common/snippets/src/lowered/pass/validate_unified_loops.cpp:62]: Discussed with @a-sidorova

File: src/common/snippets/src/lowered/pass/validate_unified_loops.cpp [hunk 6]
```diff
@@ -53,11 +53,13 @@ bool ValidateUnifiedLoops::run(LinearIR& linear_ir) {
             // if the loop has different dimension indexes, it don't have to meet the split loop related requirements
             if (dim_idx == LoopInfo::UNDEFINED_DIM_IDX)
                 continue;
-            if (std::find(dim_indexes.cbegin(), dim_indexes.cend(), dim_idx) != dim_indexes.cend()) {
-                OPENVINO_ASSERT(*dim_indexes.rbegin() == dim_idx,
-                                "Incorrect Loop ID configuration: the Loops with splitted dimension should be successively nested");
-                OPENVINO_ASSERT(loop_manager->get_loop_info(loop_ids[i - 1])->get_increment() == loop_manager->get_loop_info(id)->get_work_amount(),
-                                "Incorrect Loop ID configuration: the Loops with splitted dimension should be successively nested");
+            if (i > 0) {
+                if (std::find(dim_indexes.cbegin(), dim_indexes.cend(), dim_idx) != dim_indexes.cend()) {
```
Comment (a-sidorova) [inline src/common/snippets/src/lowered/pass/validate_unified_loops.cpp:57]:
```suggestion
                if (i > 0 && std::find(dim_indexes.cbegin(), dim_indexes.cend(), dim_idx) != dim_indexes.cend()) {
```

**Review by a-sidorova — APPROVED** [review]
Snippets part LGTM

File: src/common/low_precision_transformations/src/pad.cpp [hunk 7]
```diff
@@ -225,6 +225,10 @@ bool PadTransformation::canBeTransformed(const TransformationContext& context, s
...
             if ((beginNonZeroIdx != endNonZeroIdx) && (beginNonZeroIdx != -1) && (endNonZeroIdx != -1)) {
                 return false;
             }
```
Comment (praasz) [inline src/common/low_precision_transformations/src/pad.cpp:234]:
```suggestion
            if ((beginNonZeroIdx == -1) && (endNonZeroIdx == -1)) {
                return true;
            } 
```
?

If this new check is true then below is not reachable, but when it is false then this below is always false.

Comment (v-Golubev) [reply src/common/low_precision_transformations/src/pad.cpp:234]: When the new check is false, the check below still can be true. Let me provide an example.
`*NonZeroIdx` can be in range `[-1, pads.size() - 1]`.  Let the pads size be equal to 4. For example, `beginNonZeroIdx == 2` , `endNonZeroIdx == 3`. In this case  `((beginNonZeroIdx == -1) && (endNonZeroIdx == -1)` returns false, but `((beginNonZeroIdx != endNonZeroIdx) && (beginNonZeroIdx != -1) && (endNonZeroIdx != -1))` returns true.

Comment (praasz) [reply src/common/low_precision_transformations/src/pad.cpp:234]: True, just missed difference of using `==` and `!=`.

File: src/common/transformations/src/transformations/common_optimizations/lstm_cell_fusion.cpp (hunk 3 above)
Comment (evkotov) [reply src/common/transformations/src/transformations/common_optimizations/lstm_cell_fusion.cpp:334]: it looks ok

**Pull request merged**